### ISSUE TITLE
Triggering hover by keyboard grabs focus

### ISSUE THREAD

**Issue by maverickwoo** [Microsoft/vscode#10643]
- VSCode Version: 1.4.0
- OS Version: OS X 10.9.5

Steps to Reproduce:
1. Press Cmd-K Cmd-I on a function to trigger the hover tooltip.

After the tooltip appears, I would expect that I can start typing into the editor. However, at present I have to press Esc to kill the tooltip before I can edit again.

**Comment by alexandrudima** [comment]
This is giving focus to the hover for accessibility purposes. Otherwise, screen readers would never read the hover tooltip. I agree it might be annoying if not using a screen reader..

**Comment by maverickwoo** [comment]
You are absolutely right. I am sorry for missing this important consideration.

How about having an option such as `editor.screenreader` that defaults to true (so no change in the current behavior), but also allows some of us to declare that we are not using a screen reader? This would allow keyboard-triggered hover to behave consistently with keyboard-triggered parameter hint (Cmd-Shift-Space) (which does not grab focus).

Thank you!

**Comment by alexandrudima** [comment]
Once we adopt a newer version of Electron, we will get electron/electron#5709 which will allow us to probe if a screen reader is attached to VS Code. We could, in that case, not give focus to the hover widget.

**Comment by rbenson** [comment]
There are a range of users, sniffing for a screen reader is a bad idea. Using a mouse is quite difficult for me, and I don't need a screen reader. Having these tips are extremely helpful, especially when you consider the font size in the Intellisense or if the tip is long

**Comment by alexandrudima** [comment]
@rbenson Today, `Cmd-K Cmd-I` shows the hover and moves focus from the editor to it. Moving the focus is solely done for Screen Readers that work by reading what is focused. This issue is about detecting if a screen reader is attached, and if not, then not move focus from the editor to the hover. It is not about removing  `Cmd-K Cmd-I` entirely.

**Comment by maverickwoo** [comment]
Lately I start to notice our discussion could actually be a broader design consideration in VSCode, namely where the focus should go to when there is some kind of conceptual popup. As an example, F8 `editor.action.marker.next` would also become (imo) more convenient to non-screen-reader users if the focus can remain in the editor.

**Comment by rbenson** [comment]
@alexandrudima Thanks for the clarification. I read it as "we are planning to make `cmd-k cmd-i` work only if we can sniff a screen reader", which was a bad decision in my opinion. As long as the whole tooltip shows, on keypress.

**Comment by Ptival** [comment]
This would be a great option to have.

I have a similar issue where pressing F8 grabs the focus, and to put it back, I have to then hit Tab.
Essentially I end up always pressing Tab after F8, so I'd rather an option do this automatically. :)

**Comment by thieman** [comment]
For anyone following this issue, I've just opened a PR to attempt to implement this. Would love any feedback.
